File: api-tester/src/main/java/com/revenuecat/apitester/java/PaymentTransactionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.ProductType;
import com.revenuecat.purchases.models.PurchaseState;
import com.revenuecat.purchases.models.PurchaseType;
import com.revenuecat.purchases.models.StoreTransaction;

import org.json.JSONObject;

import java.util.List;

@SuppressWarnings({"unused"})
final class PaymentTransactionAPI {
    static void check(final StoreTransaction transaction) {
        final String orderId = transaction.getOrderId();
        final List<String> skus = transaction.getSkus();
        final List<String> productIds = transaction.getProductIds();
        final ProductType type = transaction.getType();
        final long purchaseTime = transaction.getPurchaseTime();
        final String purchaseToken = transaction.getPurchaseToken();
        final PurchaseState purchaseState = transaction.getPurchaseState();
        final Boolean autoRenewing = transaction.isAutoRenewing();
        final String signature = transaction.getSignature();
        final JSONObject originalJson = transaction.getOriginalJson();
        final String presentedOfferingIdentifier = transaction.getPresentedOfferingIdentifier();
        final String su1 = transaction.getStoreUserID();
        final PurchaseType purchaseType = transaction.getPurchaseType();
    }

    static void check(final PurchaseType type) {
        switch (type) {
            case GOOGLE_PURCHASE:
            case GOOGLE_RESTORED_PURCHASE:
            case AMAZON_PURCHASE:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PaymentTransactionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.RCProductType
import com.revenuecat.purchases.models.StoreTransaction
import com.revenuecat.purchases.models.PurchaseState
import com.revenuecat.purchases.models.PurchaseType
import org.json.JSONObject

@Suppress("unused", "UNUSED_VARIABLE")
private class PaymentTransactionAPI {
    fun check(transaction: StoreTransaction) {
        with(transaction) {
            val orderId: String? = orderId
            val skus: List<String> = skus
            val type: RCProductType = type
            val purchaseTime: Long = purchaseTime
            val purchaseToken: String = purchaseToken
            val purchaseState: PurchaseState = purchaseState
            val autoRenewing: Boolean? = isAutoRenewing
            val signature: String? = signature
            val originalJson: JSONObject = originalJson
            val presentedOfferingIdentifier: String? = presentedOfferingIdentifier
            val su1: String? = storeUserID
            val purchaseType: PurchaseType = purchaseType
        }
    }

    fun check(type: PurchaseType) {
        when (type) {
            PurchaseType.GOOGLE_PURCHASE,
            PurchaseType.GOOGLE_RESTORED_PURCHASE,
            PurchaseType.AMAZON_PURCHASE
            -> {}
        }.exhaustive
    }
}
